src/main/java/com/redhat/red/build/finder/report/TargetReport.java

```diff
@@ -0,0 +1,117 @@
+                .replaceAll("alpha", "Alpha")
+                .replaceAll("beta", "Beta")
+                .replaceAll("er", "ER")
+                .replaceAll("jb ", "JBoss ")
+                .replaceAll("jboss ", "JBoss ")
+                .replaceAll("rhel ", "RHEL ");
+
+        String newString = "";
+
+        while (it.hasNext()) {
+            String word = it.next();
+
+            if (word.length() <= 4) {
+                newString += word.toUpperCase();
+            } else {
+                newString += Character.toUpperCase(word.charAt(0)) + word.substring(1, word.length());
+            }
+
+            if (it.hasNext()) {
+                newString += " ";
+            }
+        }
+
+        prodName = newString;
+
+        return prodName;
+    }
+
+    public TargetReport(List<KojiBuild> builds) {
+        List<String> targets = builds.stream().filter(b -> b.getTaskRequest() != null && b.getTaskRequest().asBuildRequest() != null && b.getTaskRequest().asBuildRequest().getTarget() != null).map(b -> b.getTaskRequest().asBuildRequest().getTarget()).collect(Collectors.toList());
+        Map<String, Long> map = targets.stream().collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        List<Map.Entry<String, Long>> countList = map.entrySet().stream().sorted(Map.Entry.comparingByValue()).collect(Collectors.toList());
+        Collections.reverse(countList);
+
+        countList.forEach(entry -> LOGGER.info("{} {}: {}", entry.getKey(), tagNameToProductName(entry.getKey()), entry.getValue()));
+
+        LOGGER.info("Build count: {}", builds.size());
+
+        MultiValuedMap<String, KojiBuild> prodMap = new ArrayListValuedHashMap<>();
+
+        for (KojiBuild build: builds) {
+            if (build.getTaskRequest() == null || build.getTaskRequest().asBuildRequest() == null || build.getTaskRequest().asBuildRequest().getTarget() == null) {
+                  continue;
+            }
+
+            for (Entry<String, Long> countEntry : countList) {
+                String target = countEntry.getKey();
+
+                if (build.getTaskRequest().asBuildRequest().getTarget().equals(target)) {
+                    prodMap.put(target, build);
+                    break;
+                }
+            }
+        }
+
+        LOGGER.info("Product List ({}):", prodMap.asMap().keySet().size());
+
+        this.targetMap = new HashMap<>(prodMap.asMap().keySet().size());
+
+        prodMap.asMap().forEach((target, prodBuilds) -> {
+           List<String> buildList = prodBuilds.stream().map(b -> b.getBuildInfo().getNvr()).collect(Collectors.toList());
+           LOGGER.info("{}: {}", tagNameToProductName(target), buildList);
+           targetMap.put(target, buildList);
+        });
+    }
+
+    public String render() {
+        return this.targetMap.entrySet().stream().map(entry -> entry.getKey() + " " + String.join(" ", entry.getValue())).collect(Collectors.joining("\n"));
+    }
```

Comment:
Should the format between tag and builds be different (i.e. it seems to be <tag> <build1> <build2> ... ). Have RCM expressed a preference for textual list / json etc ?

Reply:
It's a space-delimited, so the design was to have the delimiter be the same. I will have to check the desired format.

Reply:
The current report now outputs HTML and not text.